### PR TITLE
Feature/issue 367

### LINKED ISSUE #367: StatisticalFactsheets: level selector
- [x] Add LevelSelector  to StatisticalFactsheets with fixed position top left --> same style than FraPlatfrom CountrySelector
- [x] navigate to level route on change 
- [x] Data should be updated

### PR DESCRIPTION
**Pull request by sorja**
Resolves #367

### CODE REVIEW

File: common/i18n/resources/en.js [hunk 1]
```diff
@@ -23,6 +23,12 @@ module.exports.translation = {
     category: {
       title: 'Select category and level',
```
Comment (minotogna) [inline common/i18n/resources/en.js:24]: this should not be there. By default selection is WO

File: server/statisticalFactsheets/api.js [hunk 2]
```diff
@@ -1,8 +1,18 @@
...
+const {
+  getStatisticalFactsheetTableAgg,
+  getStatisticalFactsheetData,
+  getStatisticalFactsheetLevelIso,
+} = require('./statisticalFactsheetsRepository')
 
 const VersionService = require('../versioning/service')
 
 module.exports.init = (app) => {
+  app.get('/statisticalFactsheets/levelIso', async (req, res) => {
```
Comment (minotogna) [inline server/statisticalFactsheets/api.js:10]: endPoint should be `/statisticalFactsheets/leveISOs`

File: server/statisticalFactsheets/statisticalFactsheetsRepository.js [hunk 3]
```diff
@@ -22,7 +22,16 @@ const getStatisticalFactsheetData = async (schemaName, level, rowNames) => {
   return camelize(result.rows)
 }
 
+const getStatisticalFactsheetLevelIso = async (schemaName) => {
```
Comment (minotogna) [inline server/statisticalFactsheets/statisticalFactsheetsRepository.js:25]: getStatisticalFactsheetLevelSOs

File: server/statisticalFactsheets/statisticalFactsheetsRepository.js [hunk 4]
```diff
@@ -22,7 +22,16 @@ const getStatisticalFactsheetData = async (schemaName, level, rowNames) => {
...
+  const result = await db.query(
+    `select distinct level from ${schemaName}.statistical_factsheets_view where level not ilike 'atlantis%' order by level`
```
Comment (minotogna) [inline server/statisticalFactsheets/statisticalFactsheetsRepository.js:27]: better exclude by level iso that starts by X

File: webapp/app/statisticalFactsheets/components/levelSelection/levelSelection.js [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+const levels = {
+  global: 'WO',
+  region: ['AF', 'AS', 'EU', 'NA', 'OC', 'SA'],
+}
+
+const LevelSelection = () => {
+  const i18n = useI18n()
+  const { levelIso } = useParams()
+
+  const { data = [], dispatch: fetchLevelIso } = useGetRequest('/api/statisticalFactsheets/levelIso')
```
Comment (minotogna) [inline webapp/app/statisticalFactsheets/components/levelSelection/levelSelection.js:23]: better to call this `fetchLevelISOs`

File: webapp/app/statisticalFactsheets/components/levelSelection/levelSelection.js [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+  useEffect(() => {
+    fetchLevelIso()
+    window.addEventListener('click', outsideClick)
+
+    return () => {
+      window.removeEventListener('click', outsideClick)
+    }
+  }, [])
+
+  const getLabel = () => {
```
Comment (minotogna) [inline webapp/app/statisticalFactsheets/components/levelSelection/levelSelection.js:41]: better to have a constant. no need for function.
also, do u run in a hooks change order warning?

File: webapp/app/statisticalFactsheets/components/levelSelection/levelSelection.less [hunk 7]
```diff
@@ -0,0 +1,11 @@
+@import (reference) 'webapp/main/styles/config';
+
+.level-selection {
+  position: fixed;
+  left: 0;
+  top: 0;
+  width: 400px;
+  height: 61px;
+  min-height: 61px;
```
Comment (minotogna) [inline webapp/app/statisticalFactsheets/components/levelSelection/levelSelection.less:9]: is there a need for both?

File: webapp/app/statisticalFactsheets/components/levelSelection/list.js [hunk 8]
```diff
@@ -0,0 +1,64 @@
...
+import { useI18n } from '@webapp/components/hooks'
+import * as BasePaths from '@webapp/main/basePaths'
+import { Link } from 'react-router-dom'
+import { useSelector } from 'react-redux'
+import * as CountryState from '@webapp/app/country/countryState'
+
+const levels = {
+  global: 'WO',
+  region: ['AF', 'AS', 'EU', 'NA', 'OC', 'SA'],
+}
```
Comment (minotogna) [inline webapp/app/statisticalFactsheets/components/levelSelection/list.js:14]: this is duplicated. Create a common object call Levels

File: webapp/app/statisticalFactsheets/components/levelSelection/list.js [hunk 9]
```diff
@@ -0,0 +1,64 @@
...
+  const { item } = props
+  const i18n = useI18n()
+  const getLabel = (key) => {
+    if (!key) return null
+    if (key === levels.global || levels.region.includes(key)) {
+      return i18n.t(`statisticalFactsheets.category.${key}`)
+    }
+    const country = useSelector(CountryState.getCountryByCountryIso(key))
+    return country ? country.listName[i18n.language] : null
+  }
```
Comment (minotogna) [inline webapp/app/statisticalFactsheets/components/levelSelection/list.js:26]: this is also duplicated. move the code to a common place.
Duplicated code leads to bugs

File: webapp/app/statisticalFactsheets/components/levelSelection/list.js [hunk 10]
```diff
@@ -0,0 +1,64 @@
...
+
+  return (
+    <Link to={BasePaths.getStatisticalFactsheetsWithLevelIso(item)} className="country-selection-list__row">
+      <span className="country-selection-list__primary-col">{getLabel(item)}</span>
+    </Link>
+  )
+}
+
+ListItem.propTypes = {
```
Comment (minotogna) [inline webapp/app/statisticalFactsheets/components/levelSelection/list.js:35]: move ListItem to a separate file.

File: webapp/app/statisticalFactsheets/statisticalFactsheets.js [hunk 11]
```diff
@@ -10,13 +13,21 @@ import ForestOwnership from './ForestOwnership'
 import ForestAreaWithinProtectedAreas from './ForestAreaWithinProtectedAreas'
 import NaturallyRegeneratingForest from './NaturallyRegeneratingForest'
 import PrimaryDesignatedManagementObjective from './PrimaryDesignatedManagementObjective'
+import LevelSelection from './components/levelSelection'
+
+const defaultLevelIso = 'WO'
```
Comment (minotogna) [inline webapp/app/statisticalFactsheets/statisticalFactsheets.js:18]: this is duplicated. no use of constant. use common Level object

**Comment by sorja** [comment]
> just checked the branch out.
> 
> ```
> ERROR in ./webapp/app/statisticalFactsheets/components/chart/Chart.js
> Module not found: Error: Can't resolve 'chart.js' in '.../fra-platform/webapp/app/statisticalFactsheets/components/chart'
> ```

Did you install npm packages?

File: common/i18n/resources/en.js [hunk 12]
```diff
@@ -21,8 +21,13 @@ module.exports.translation = {
...
   statisticalFactsheets: {
     category: {
-      title: 'Select category and level',
       WO: 'World',
+      AF: 'Africa',
+      AS: 'Asia',
+      EU: 'Europe',
+      NA: 'North America',
+      OC: 'Oceania',
+      SA: 'South America',
```
Comment (minotogna) [inline common/i18n/resources/en.js:30]: please remove this. use instead area.${levelIso}.listName (see my pull request)

**Pull request merged**